launch_ros/launch_ros/actions/load_composable_nodes.py

```diff
@@ -230,19 +233,48 @@ def execute(
         # Generate load requests before execute() exits to avoid race with context changing
         # due to scope change (e.g. if loading nodes from within a GroupAction).
         load_node_requests = []
+        autostart_actions = []
         for node_description in self.__composable_node_descriptions:
             request = get_composable_node_load_request(node_description, context)
             # The request can be None if the node description's condition evaluates to False
             if request is not None:
                 load_node_requests.append(request)
 
+            # If autostart is enabled, transition to the 'active' state.
+            if node_description.node_autostart:
+                complete_node_name = request.node_namespace + request.node_name
+                if not complete_node_name.startswith('/'):
+                    complete_node_name = '/' + complete_node_name
+                self.__logger.info(
+                    'Autostart enabled for requested lifecycle node {}'.format(complete_node_name))
+                node_description.init_lifecycle_event_manager(node_description, context)
```

Comment:
This is related to my comment about an `on_execute` "virtual" function on the composable node classes, but it's weird to be passing itself into this function. I think this could just be:

```suggestion
                node_description.init_lifecycle_event_manager(context)
```

Reply:
I agree, this is an artifact of moving the logic from lifecycle node to the composition node. 

Completed